### DISCUSSION TITLE
osu crashes, terminal output gives some output that seems related to realm

### DISCUSSION THREAD

**Discussion by rpst39** [ppy/osu#25564] [Q&A]
### Type

Crash to desktop

### Bug description

I haven't played osu in a while so downloaded the latest release but I cannot launch the game. I tried previous releases too but I just can't launch the game. I have the osu folder in a exfat partition to have it accessible on both macOS and linux, and have this issue on both of them. The only log file that gets modified is the runtime log file, the rest are not modified. Other logs are from the last successful start and that happened when I was booted at macOS so that's why the os is not matching in the logs. The runtime log file was created on linux. 

This is the output I get when I run the appimage file in a terminal:

`/home/runner/work/realm-dotnet/realm-dotnet/wrappers/realm-core/src/realm/index_string.hpp:364: [realm-core-13.20.1] Assertion failed: Array::get_context_flag_from_header(alloc.translate(ref)) with (ref, size_t(alloc.translate(ref))) =  [1363416, 139925539573208]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x55ee9c) [0x7f430975ee9c]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x3cff37) [0x7f43095cff37]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x4c5ab9) [0x7f43096c5ab9]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x52142f) [0x7f430972142f]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x4b1ac6) [0x7f43096b1ac6]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x4b1e63) [0x7f43096b1e63]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x29ea37) [0x7f430949ea37]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x29eb20) [0x7f430949eb20]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x2a339b) [0x7f43094a339b]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x2db5de) [0x7f43094db5de]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x5b3dee) [0x7f43097b3dee]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x5b40d1) [0x7f43097b40d1]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x2f8e40) [0x7f43094f8e40]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x2db39e) [0x7f43094db39e]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(+0x2748b5) [0x7f43094748b5]
/tmp/.mount_osu.ApU8oF6k/usr/bin/librealm-wrappers.so(shared_realm_open+0x28) [0x7f4309475b28]
[0x7f43c832a896]
!!! IMPORTANT: Please report this at https://github.com/realm/realm-core/issues/new/choose
Thread name: osu!Aborted (core dumped)
`

I think it has something to do with my folder as when I rename it and create another folder with the same name osu launches fine and starts fresh (says that folder is empty and asks if I want to still use it but that is normal behavior). Could it be that files in my folder corrupted?

### Version

2023.1114.1

### Logs

[runtime.log](https://github.com/ppy/osu/files/13453132/runtime.log)
[performance.log](https://github.com/ppy/osu/files/13453178/performance.log)
[network.log](https://github.com/ppy/osu/files/13453180/network.log)
[database.log](https://github.com/ppy/osu/files/13453181/database.log)

**Comment by bdach** [comment]
Do things work normally if your files are not on an exFAT partition? We've historically had multiple issues reported by users who attempted to store game files on one.

Although at some point I'm pretty sure we were in a place that supported that, I'm not sure what the current state of that is. Something may have regressed realm side.

**Comment by rpst39** [comment]
Unfortunately that didn't fix the issue, it is still doing the exact same thing.

**Comment by bdach** [comment]
What linux distribution is this?

**Reply by rpst39** [reply]
Arch linux

**Comment by rpst39** [comment]
Okay so I don't think this was an issue related to osu as even when I started fresh I had some weird issues so I decided to check my ram and memtest86 gave me some errors which got fixed by disabling the overclock on it. I think the folder somehow got corrupted due to the ram errors.
